cocotb/regression.py

```diff
@@ -406,9 +400,16 @@ def _result_was():
 
         return result_pass, sim_failed
 
-    def execute(self):
-        self._running_test = cocotb.regression_manager.next_test()
-        if self._running_test:
+    def execute(self) -> None:
+        while True:
```

Comment:
This doesn't look like it should be looped, unless `add_test` has suddenly become blocking

Reply:
Nevermind, I now understand: see my below comment.